src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -5495,6 +5527,332 @@ register_handler(int sig, struct sigaction *old_handler)
 	}
 }
 
+static int
+allocate_aio_ctx(io_context_t ctx, int depth)
+{
+	int i;
+
+	pthread_mutex_lock(&lock_aio);
+	if (aio_ctx_count >= MAX_AIO_CTX)
```

Comment:
do we are limiting to 8? why?

Reply:
I am not familiar to the use cases of libaio. Not sure how large the limit should be. Sure. We can bump up the default limit and revise the code to adapt larger limit. We could set an env variable to allow user to modify the limit.

Reply:
my question was why you set this to 8 and if i understood your answer, it was just random?
I think this is something that needs more investigation before it can be set it seems.

Reply:
Yes. I picked it randomly in the preliminary implementation.

Reply:
Currently separate EQs are used for each aio context, so I randomly picked a small number.